main_f4.c

```diff
@@ -715,6 +715,14 @@ main(void)
 
 #endif
 
+#define BL_DATA ((uint8_t *)(0x08000000 + 0x4000 - 16)) // last 16 bytes of block
```

Comment:
I think we want this inside an #ifdef block, with that set in hw_config.h
my concern is this bootloader is used for a lot of different boards, and if we enable this globally then I'd have to test all of them to make sure it doesn't do any harm
If we use a #ifdef to enable it then I could put it in now and know its safe

Reply:
done